src/core/server/saved_objects/migrationsv2/initial_state.ts

```diff
@@ -79,5 +86,6 @@ export const createInitialState = ({
     logs: [],
     unusedTypesQuery: excludeUnusedTypesQuery,
```

Comment:
We're now excluding specific types, excluding documents that satisfy the `excludeFilterHooks` and migrations per SO type. In short, where we _used_ to have one way of handling document transforms between versions (migrations), we now have three subtly different methods each doing a different job. Should we be at all concerned about the amount of cleanup we're having to do lately?

Reply:
I definitely think that as a follow up to this we should remove the `unusedTypeQuery` and instead move those filters into the SO type registrations for these unused types. That will consolidate these different ways of doing things to just two:
- Register migrations for an SO type for schema transformations
- Register a `deleteOnUpgrade` filter for removing unneeded documents

We can then add a 'sugar' API for removing types like `savedObjects.registerRemovedType('my-type')` that essentially just does this under the hood:

```ts
registerRemovedType(typeName: string, indexPattern?: string) {
  core.savedObjects.registerType({
    name: typeName,
    indexPattern,
    // completely disable mappings
    mappings: null,
    // remove all from upgrade
    deleteOnUpgrade: () => ({ bool: { must: { term: { type: typeName } } } })
  })
}
```

Reply:
> I definitely think that as a follow up to this we should remove the unusedTypeQuery and instead move those filters into the SO type registrations for these unused types.

Agree, is there a reason we don't use `excludeUnusedTypesQuery` as the default value in 
https://github.com/elastic/kibana/pull/106534/files#diff-d2fe0f4c46b3cffd55e6c042125be7bbff5930f8a7a8cb67adf72e7f81c6c325R47?